Remove Hardcoded Routes/Endpoints (CRASM-3291)

hawkishpolicy opened the pull request:
<!-- GitHub renders PRs such that soft line breaks are treated as hard
line breaks.  In order to make this template render as expected we
therefore have to avoid soft breaks and therefore will offend
markdownlint with long lines.  This is the reason for the markdownline
disable directive just below.

For more details see:
https://github.com/github/markup/issues/1050#issuecomment-294654762 -->
<!-- markdownlint-disable MD013 -->
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##
- Replaced all hard coded route and endpoint strings throughout the frontend directory with references to centralized constants stored in either ROUTES or ENDPOINTS.
   - Routing example: 
      - '/VSDashboard' is now ROUTES.VSDASHBOARD
   - Endpoint example:
      - `/stats` is now ENDPOINTS.STATS
- Placeholders are now standardized:
  - Routes use :placeholder
  - Endpoints use {placeholder}
- The replace prototype method is used to insert relevant data in place of the placeholder.
  - Routing example for 'inventory/domain/:domainId
     - ROUTES.DOMAIN.replace(':domainId', cellValues.row.id)
  - Endpoint example for '/domain/{domain_id}'
     - ENDPOINTS.DOMAIN.replace('{domain_id}', domainId)         
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##
- Standardizes and centralizes all routes and endpoints throughout the frontend directory.
- Closes CRASM-32981
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##
- tested locally.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [ ] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.
- [ ] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).
- [ ] Create a pre-release (necessary if and only if the pre-release version was bumped).

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Create a release (necessary if and only if the version was bumped).

hawkishpolicy committed:
Initial commit for removing hard-coded routes

- Created constants for API endpoints and frontend routes.
- Updated Routes.tsx to use the new route constants.
- Began applying endpoint constants in API calls throughtout the frontend.

hawkishpolicy committed:
Add remaining endpoints to constants file:

- Brought over remaining hard-coded endpoints to `frontend/src/constants/endpoints.ts`.
- Continued refactoring code to use these constants instead of hard-coded strings.
- Made some minor edits for endpoint consistency and readability.

hawkishpolicy committed:
Apply ENDPOINT constants across the following:

- DrawerInterior.tsx
- AppGate.tsx
- ExportCustomerMetricsButton.tsx
- ScansWidget.tsx
- ScanForm.tsx
- useFetchNotification.ts
- useMaintenanceNotifications.ts
- AuthLogin.tsx
- OktaCallback.tsx
- Organization.tsx
- OrgMembers.tsx
- OrgScanHistory.tsx
- OrgSettings.tsx
- OrganizationForm.tsx

hawkishpolicy committed:
Update endpoints in the following files:
- frontend/src/pages/RegionUsers/RegionUsers.tsx
- frontend/src/pages/Scan/Scan.tsx

Cleaned up code in:
- frontend/src/pages/Organization/OrgSettings.tsx

hawkishpolicy committed:
Update routes in the following files:

- added ROUTES.SCANS and ROUTES.SCANS_HISTORY to frontend/src/constants/routes.ts
- replaced hard-coded '/scans' and '/scans/history' paths with ROUTES.SCANS and ROUTES.SCANS_HISTORY in frontend/src/pages/Scans/Scans.tsx

hawkishpolicy committed:
Update endpoints in the following files:

- frontend/src/pages/Scans/ScansView.tsx
- frontend/src/pages/Scans/ScanTasksView.tsx

hawkishpolicy committed:
Update domain detail route usage in ResultCard.tsx

- Replace hardcoded route with ROUTES.DOMAIN

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Remove-Frontend-Hard-Coded-Routes-CRASM-3291

hawkishpolicy committed:
Update api calls to use constants in UserForm.tsx

hawkishpolicy committed:
Update API calls to use constants in Users.tsx

hawkishpolicy committed:
Update the following to use ROUTES constants:

- Vulnerabilities.tsx
- OverviewSection.tsx
- TopVulnerableHosts.tsx

hawkishpolicy committed:
Update the following to use route constants:

- AuthRoute.tsx
- InfoLabel.tsx
- FilterDrawerV2.tsx
- Layout.tsx
- RouteGuard.tsx
- Vulnerability.tsx

hawkishpolicy committed:
Update API calls to constants in the following:

- RegionAndOrganizationFilters.tsx
- VSDashRegionAndOrgFilters.tsx

hawkishpolicy committed:
Fix placeholder typo in UserForm.tsx

hawkishpolicy committed:
Update the following to use route constants:

- frontend/src/components/FindingsLibrary/FindingsHeader.tsx

hawkishpolicy committed:
Update the following to use ROUTES constants:

- Subnav.tsx
- NavMenuButton.tsx

hawkishpolicy committed:
Update the following to use route constants:

- RouteGuard.tsx
- filterPaths.ts
- NavigationContext.ts
- cleaned up code in ScanTasksView.tsx

hawkishpolicy committed:
Update the following to use endpoint constants:

- SavedSearchContextProvider.tsx
- StaticsContextProvider.tsx

hawkishpolicy committed:
Update the following to use route cpnstants:

- Domain/DomainDetails.tsx
- Domains/DomainDetails.tsx
- Domains/Domains.tsx

hawkishpolicy committed:
Update the following to use ROUTES constants:

- LoginGovCallback.tsx
- OktaCallback.tsx

hawkishpolicy committed:
Update the following to use ROUTES constants:

- Organization.tsx
- Organizations.tsx

hawkishpolicy committed:
Update Organizations.tsx to use ENDPOINT constants

hawkishpolicy committed:
Update UserForm.tsx to use ENDPOINTS constants

hawkishpolicy committed:
Update Vulns.tsx to use ENDPOINTS constants

hawkishpolicy committed:
Update the following to use ENDPOINTS constants:

- QueueMonitorView.tsx
- Inventory.tsx

hawkishpolicy committed:
Update ScansView to use ENDPOINTS constant

hawkishpolicy committed:
Update the following to use ROUTES constants:

- KeyMetrics.tsx
- SeverityByProminence.tsx
- TopVulnerabilities.tsx

Commit:
Fix vs_port and vs_vuln scans to allow local generation

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Remove-Frontend-Hard-Coded-Routes-CRASM-3291

hawkishpolicy committed:
Merge remote-tracking branch 'origin/fix-this-hoopla' into Remove-Frontend-Hard-Coded-Routes-CRASM-3291

hawkishpolicy committed:
Update the following to use ROUTES constants:

- detected-host-widget.spec.ts

hawkishpolicy committed:
Update termsOfUse.spec.tsx to use ENDPOINTS const

hawkishpolicy committed:
Update the following to use ROUTES constants:

- detected-hosts-widget.spec.ts

hawkishpolicy committed:
Update the following to use ROUTES constants:

- detected-vulnerabilities-widget.spec.ts

hawkishpolicy committed:
Update the following to use ROUTES constants:

- domains-table-sorting.spec.ts

hawkishpolicy committed:
Update the following to use ROUTES constants:

- home.spec.ts

hawkishpolicy committed:
Update inventory.spec.ts to use ROUTES constants

hawkishpolicy committed:
Update the following to use ROUTES constants:

- latest_scanning_summary_widget.spec.ts

hawkishpolicy committed:
Update vulnerabilities.spec.ts to use ROUTES const

hawkishpolicy committed:
Fix route typo in ResultCard.tsx

hawkishpolicy committed:
Create MATOMO route constant and apply it

hawkishpolicy committed:
Setup Playwright:

- installed playwright
- added .env variables for playwright
- .auth files were created automatically
- setup 2FA for the 4 user types.

hawkishpolicy committed:
Setup playwright testing:

- Installed playwright.
- Added .env variables for playwright.
- setup 2FA authentication for global-admin, global-view, regional-admin, and standard-user roles.
- .auth files created automatically after successful login.

jsalinasnttdata reviewed (dismissed):
A lot of changes! Everything looks good to me. I also tested locally and everything works as intended.

One thing I noticed is that the backend changes to temporarily fix the data issue is committed in this repo. Should this be removed as it's already being worked on in a different ticket?

hawkishpolicy commented:
> A lot of changes! Everything looks good to me. I also tested locally and everything works as intended.
> 
> One thing I noticed is that the backend changes to temporarily fix the data issue is committed in this repo. Should this be removed as it's already being worked on in a different ticket?

I don't know. Great point. I'll ask in the review. I don't think it will cause a merge conflict if the other PR is merged first.

hawkishpolicy committed:
Remove Janson's fix for local vs summary creation:

- Reverted files to their previous state before the fix.
- His fix is coming in a different PR.

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Remove-Frontend-Hard-Coded-Routes-CRASM-3291

hawkishpolicy commented:
> A lot of changes! Everything looks good to me. I also tested locally and everything works as intended.
> 
> One thing I noticed is that the backend changes to temporarily fix the data issue is committed in this repo. Should this be removed as it's already being worked on in a different ticket?

Removed his hot fix via: 
commit a14c4e25b0f1425db51d5e686b14d5eaf248d476

ameliav reviewed (changes requested):
Remove the playwright/.auth folder contents

hawkishpolicy committed:
Remove .auth files from Playwright setup:

- they should be on .git ignore

ameliav reviewed (dismissed):
Looks good. UAT should test all routes and endpoints to verify nothing broke.

Please do not merge this until the following PRs are merged into develop:
https://github.com/cisagov/XFD/pull/1282
https://github.com/cisagov/XFD/pull/1283

hawkishpolicy commented:
> Remove the playwright/.auth folder contents

Removed via [d040443](https://github.com/cisagov/XFD/pull/1291/commits/d040443d49b975863fe71f444c30702a7ea25f3d)

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Remove-Frontend-Hard-Coded-Routes-CRASM-3291

hawkishpolicy committed:
Remove TERMS constant

- No longer needed after Terms refactor.

RKohler21 committed:
Terms of Use Appgate Updates

RKohler21 committed:
Merge branch 'TermsOfUse_Refactor-3304' into Remove-Frontend-Hard-Coded-Routes-CRASM-3291

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into Remove-Frontend-Hard-Coded-Routes-CRASM-3291

hawkishpolicy committed:
Remove console logs from TermsGate component

hawkishpolicy committed:
Remove unused location imports from TermsGate

ameliav reviewed (dismissed):
Re-reviewed and tested new changes. LGTM!

RKohler21 commented:
Reviewed and tested, everything looks good from my end.

hawkishpolicy committed:
Standardize imports aliases:

- Change imports from 'constants/...' to '@/constants/...'

Applied ROUTES.NOTIFCATION to delete notification action.
 # Please enter the commit message for your changes. Lines starting

hawkishpolicy committed:
Remove throw error from useFirstLoginPopup hook

hawkishpolicy committed:
Edit isDrillDownDestination:

- stripped parameter segments from ROUTES.VULNERABILITY and ROUTES.DOMAIN before checking
- removed double slashes and trailing slashes for accurate comparison

hawkishpolicy committed:
Add minor fixes to route matching logic & imports:

- Replaced hard-coded '/logout' check with ROUTES.LOGOUT constant.
- Removed duplicate 'search/organizations' endpoint.
- Improved path matching logic for vulnerability and domain routes to ensure accurate segment matching in the isDrillDownDestination function.
